### PR TITLE
Feat: Port REI 5 to REI 6

### PR DESCRIPTION
**Pull request by AlphaMode**

### CODE REVIEW

File: src/main/java/dev/galacticraft/mod/compat/rei/client/GalacticraftREIClientPlugin.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        for (Item item : GalacticraftItem.HIDDEN_ITEMS) {
+            registry.removeEntry(EntryStacks.of(item));
+        }
+    }
+
+    @Override
+    public void registerExclusionZones(ExclusionZones zones) {
+        zones.register(MachineHandledScreen.class, provider -> {
+            MachineHandledScreen<?,?> machineScreen = provider;
+            List<Rectangle> l = new ArrayList<>();
```
Comment (JoeZwet) [inline src/main/java/dev/galacticraft/mod/compat/rei/client/GalacticraftREIClientPlugin.java:90]: use proper variable names

File: src/main/java/dev/galacticraft/mod/compat/rei/client/category/DefaultCompressingCategory.java [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * @author <a href="https://github.com/TeamGalacticraft">TeamGalacticraft</a>
+ */
+@Environment(EnvType.CLIENT)
+public class DefaultCompressingCategory implements DisplayCategory<DefaultCompressingDisplay> {
+    private static final Identifier DISPLAY_TEXTURE = new Identifier(Constant.MOD_ID, "textures/gui/rei_display.png");
```
Comment (JoeZwet) [inline src/main/java/dev/galacticraft/mod/compat/rei/client/category/DefaultCompressingCategory.java:62]: move this to `Constant.ScreenTexture`

**Review by JoeZwet — CHANGES_REQUESTED** [review]
couple changes, will test on my local when i get home

File: src/main/java/dev/galacticraft/mod/compat/rei/client/category/DefaultCompressingCategory.java [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+                int width = MathHelper.ceil((double) (System.currentTimeMillis() / 250L) % 24.0D);
+                this.drawTexture(matrices, startPoint.x + 24, startPoint.y + 18, 82, 91, width, 17);
+            }
+
+            @Override
+            public List<? extends Element> children() {
+                return Collections.emptyList();
+            }
+        }
+
```
Comment (marcus8448) [inline src/main/java/dev/galacticraft/mod/compat/rei/client/category/DefaultCompressingCategory.java:101]: Do we want to export this to its own class or as an inner class?

Comment (AlphaMode) [reply src/main/java/dev/galacticraft/mod/compat/rei/client/category/DefaultCompressingCategory.java:101]: Tbh we might not even need it

Comment (JoeZwet) [reply src/main/java/dev/galacticraft/mod/compat/rei/client/category/DefaultCompressingCategory.java]: done in 84de190

**Review by marcus8448 — APPROVED** [review]
Code LGTM
(I haven't tested it)

**Pull request merged**